.github/workflows/detect-jobs-to-run.js

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env node
+// @ts-check
+
+// This was originally done in prisma/prisma repository and adapted to e2e
+// https://github.com/prisma/prisma/blob/main/.github/workflows/detect-jobs-to-run.js
+
+const { stdin } = process
+
+// From https://github.com/sindresorhus/get-stdin/blob/main/index.js
+async function getStdin() {
+  let result = ''
+
+  if (stdin.isTTY) {
+    return result
+  }
+
+  stdin.setEncoding('utf8')
+
+  for await (const chunk of stdin) {
+    result += chunk
+  }
+
+  return result
+}
+
+async function main() {
+  const testDirectories = [
+    'process-managers',
+    'docker',
+    'core-features',
+    'migrate',
+    'engine',
+    'os',
+    'node',
+    'binaries',
+    'packagers',
+    'frameworks',
+    'platforms',
+    'paltforms-serverless',
+    'workers',
+    'bundlers',
+    'libraries',
+    'databases',
+    'databases-macos',
+    'test-runners',
+  ]
```

Comment:
If in any way possible we should avoid duplicating this list. Maybe just filter out the folders that are _not_ tests? That should be more static.

Reply:
Yep maybe doing the same as in 
https://github.com/prisma/e2e-tests/blob/60f3b664ca7b66e63904c266528bf3e2a3f3ef58/.github/scripts/check-folders.js

```
  const jobKeys = Object.keys(optionalTestYaml['jobs'])
    .concat(Object.keys(testYaml['jobs']))
    .filter...
```